scipy/optimize/_qap.py

```diff
@@ -0,0 +1,418 @@
+    vertices of two graphs which minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2]_.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square, non-negative
+        A square adjacency matrix
+
+    dist_matrix : 2d-array, square, non-negative
+        A square adjacency matrix
+
+    seed : 2d-array, optional, (default = None)
+        Allows the user apply a seed, fixing part of the matching between
+        the two adjacency matrices.
+        For column 1, each entry is an index of a node in `cost_matrix`.
+        For column 2, each entry is an index of a node in `dist_matrix`.
+        The elements of ``seed[:, 0]`` and ``seed[:, 1]`` are vertices
+        which are known to be matched, that is, `seed[i, 0]` is matched to
+        vertex `seed[i, 1]`. Array shape (m , 2) where m <= number of nodes
+
+    maximize : bool (default = False)
+        Gives users the option to solve the Graph Matching Problem (GMP)
+        rather than QAP. This is accomplished through trivial negation
+        of the objective function.
+
+
+    options : dict, optional
+        A dictionary of solver options. All methods accept the following
+        options:
+
+            n_init : int, positive (default = 1)
+                Number of random initializations of the starting
+                permutation matrix that the FAQ algorithm will undergo.
+            init_method : string (default = 'barycenter')
+                The algorithm may be sensitive to the initial permutation
+                matrix (or search position) chosen due to the possibility
+                of several local minima within the feasible region.
+                With only 1 initialization, a barycenter init will
+                likely return a more accurate permutation.
+                Choosing several random initializations as opposed to
+                the non-informative barycenter will likely result in a
+                more accurate result at the cost of higher runtime.
+                The initial position chosen:
+                "barycenter" : the non-informative "flat doubly stochastic
+                matrix,":math:`J=1*1^T /n` , i.e the barycenter of the
+                feasible region (where n is the number of nodes and '1' is
+                a (n, 1) array of ones)
+                "rand" : some random point near :math:`J, (J+K)/2`, where K
+                is some random doubly stochastic matrix
+            max_iter : int, positive (default = 30)
```

Comment:
```suggestion
            maxiter : int, positive (default = 30)
```
For consistency with `minimize`:
![image](https://user-images.githubusercontent.com/6570539/82103557-544a4300-96c8-11ea-8360-ecbbcd2d7467.png)

`linprog`:
![image](https://user-images.githubusercontent.com/6570539/82103577-60ce9b80-96c8-11ea-8c21-bd357e3db132.png)